tests/kokkos/kokkos_utils_tests.cpp

```diff
@@ -194,4 +196,106 @@ TEST_CASE("team_utils_large_ni", "[kokkos_utils]")
   test_utils_large_ni(.5);
 }
 
+template<typename Scalar, int n_teams, int work_per_team>
+void test_parallel_reduce()
+{
+  using Device = ekat::DefaultDevice;
+  using MemberType = typename ekat::KokkosTypes<Device>::MemberType;
+  using ExeSpace = typename ekat::KokkosTypes<Device>::ExeSpace;
+
+#ifdef KOKKOS_ENABLE_OPENMP
+  const int n = omp_get_max_threads();
+  // test will not work with more than 16 threads
+  if (n > 16)
+  {
+    WARN("Skipped because this test doesn't support more than 16 threads");
+    return;
+  }
+#endif
+
+  const int length = n_teams*work_per_team;
+  Kokkos::View<Scalar*, ExeSpace> data("data", length);
+  const auto data_h = Kokkos::create_mirror_view(data);
+  auto raw = data_h.data();
+  for (int i = 0; i < length; ++i)
+    raw[i] = 1.0/(i+1);
+  Kokkos::deep_copy(data, data_h);
+
+  Kokkos::View<Scalar*> results ("results", n_teams);
+  const auto results_h = Kokkos::create_mirror_view(results);
+
+  std::vector<Real> serial_results(n_teams);
+  for (unsigned int t=0; t<n_teams; ++t)
+    for (unsigned int w=0; w<work_per_team; ++w)
+    {
+      const int i = t*work_per_team + w;
+      serial_results[t] += 1.0/(i+1);
+    }
+
+  const auto policy =
+    ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(n_teams, work_per_team);
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+    const int t = team.league_rank();
+    Scalar team_result = Scalar();
+
+    const int begin = t*work_per_team;
+    const int end = begin + work_per_team;
+    ekat::ExeSpaceUtils<ExeSpace>::parallel_reduce(team, begin, end,
+        [&] (const int k, Scalar& reduction_value) {
+                 reduction_value += data[k];
+               }, true/*serialize*/, team_result);
```

Comment:
You might consider adding a short "fake" test for serialize=false. Not to check the answer (will be non deterministic), but to check stuff compiles and runs without issues.

Reply:
On this test, I added a template parameter for whether or not the `REQUIRE(results_h(0) == serial_result);` is run based on `EKAT_TEST_STRICT_FP`. So on a non-bfb run the test runs just without the failing assertion. Is this acceptable? I'm can definitely change this.

Reply:
I am not a fan of `EKAT_TEST_STRICT_FP`. Ekat offers some features, that should work (perhaps following different code paths) regardless. We should test both a "strict" code flow, as well as a "release" code flow. Obviously, the checks that need to occur are different (like you suggested, not running the REQUIRE clause if not a strict run).

That said, I don't see where the extra tepmlate parameter is. Maybe you haven't pushed yet? Either way, let's make sure that the function is executed with both serialize true and false.